<launch>
    
  <!-- 
  This launcher is used to record rfid data on Linda.
  
  Check tag poses!!! 
  -->
  <arg name="robot_tf" default="odom"/>
  <arg name="world_tf" value="map"/>
  <!--  RFID reader transmitted power in 100 * dBm --> 
  <arg name="txPower100dBm" value="2000"/>
  
  
  
  <!--  Robot driver --> 
  <include file="$(find strands_bringup)/launch/strands_robot.launch">
        <arg name="with_mux" value="false"/>
  </include>

  <!--  Mongo  --> 
  <include file="$(find mongodb_store)/launch/mongodb_store.launch">
        <arg name="db_path" value="/opt/strands/aaf_datacentre"/>
  </include>
 
  <!--  Topo Nav  --> 
  <include file="$(find ff_bringup)/launch/ff_navigation.launch">
        <arg name="map" value="/localhome/lcas/Maps/fab-lab-rfid.yaml"/>
        <arg name="topological_map" value="fab-lab-rfid"/>
  </include>

  <!-- rfid reader -->
  <node pkg="rfid_node" type="rfid_reader_ros.py" name="rfid_node" output="screen" respawn="true">
      <param name="RFIDdeviceURI" value="tmr:///dev/ttyACM0"/>
      <param name="txpower" value="$(arg txPower100dBm)"/>
  </node>
  
  <!--  Tag positions referred to map coordinates  --> 
  <include file="$(find pr_model)/launch/tagData.launch">
      <arg name="robot_tf" value="$(arg robot_tf)"/>
      <arg name="world_tf" value="$(arg world_tf)"/>
      
      <arg name="tagID" value="390100010000000000000001"/>
      <arg name="tag_num" value="1"/>
      <arg name="worldTagPose" value="0.161 -1.803 0.0 0.0 0.0 0.729 0.685"/>
  </include>
  
  <include file="$(find pr_model)/launch/tagData.launch">
      <arg name="robot_tf" value="$(arg robot_tf)"/>
      <arg name="world_tf" value="$(arg world_tf)"/>
      
      <arg name="tagID" value="390100010000000000000002"/>
      <arg name="tag_num" value="2"/>
      <arg name="worldTagPose" value="-0.926 -2.014 0.0 0.0 0.0 0.719 0.695"/>
  </include>
  
  <include file="$(find pr_model)/launch/tagData.launch">
      <arg name="robot_tf" value="$(arg robot_tf)"/>
      <arg name="world_tf" value="$(arg world_tf)"/>
      
      <arg name="tagID" value="390100010000000000000003"/>
      <arg name="tag_num" value="3"/>
      <arg name="worldTagPose" value="-2.219 -0.996 0.0 0.0 0.0 0.062 0.998"/>
  </include>
  
  <include file="$(find pr_model)/launch/tagData.launch">
      <arg name="robot_tf" value="$(arg robot_tf)"/>
      <arg name="world_tf" value="$(arg world_tf)"/>
      
      <arg name="tagID" value="390100010000000000000004"/>
      <arg name="tag_num" value="4"/>
      <arg name="worldTagPose" value="-2.192 0.245 0.0 0.0 0.0 0.037 0.999"/>
  </include>
  
  <include file="$(find pr_model)/launch/tagData.launch">
      <arg name="robot_tf" value="$(arg robot_tf)"/>
      <arg name="world_tf" value="$(arg world_tf)"/>
      
      <arg name="tagID" value="390100010000000000000005"/>
      <arg name="tag_num" value="5"/>
      <arg name="worldTagPose" value="-2.546 1.443 0.0 0.0 0.0 0.053 0.999"/>
  </include>
  
  <include file="$(find pr_model)/launch/tagData.launch">
      <arg name="robot_tf" value="$(arg robot_tf)"/>
      <arg name="world_tf" value="$(arg world_tf)"/>
      
      <arg name="tagID" value="390100010000000000000006"/>
      <arg name="tag_num" value="6"/>
      <arg name="worldTagPose" value="-2.126 3.746 0.0 0.0 0.0 -0.484 0.875"/>
  </include>
  
  <include file="$(find pr_model)/launch/tagData.launch">
      <arg name="robot_tf" value="$(arg robot_tf)"/>
      <arg name="world_tf" value="$(arg world_tf)"/>
      
      <arg name="tagID" value="390100010000000000000007"/>
      <arg name="tag_num" value="7"/>
      <arg name="worldTagPose" value="-0.439 4.765 0.0 0.0 0.0 -0.484 0.875"/>
  </include>
  
      <include file="$(find pr_model)/launch/tagData.launch">
      <arg name="robot_tf" value="$(arg robot_tf)"/>
      <arg name="world_tf" value="$(arg world_tf)"/>
      
      <arg name="tagID" value="390100010000000000000008"/>
      <arg name="tag_num" value="8"/>
      <arg name="worldTagPose" value="1.498  2.975  0.000 0.000 0.000 0.969 -0.248"/>
  </include>     
</launch>
